File: Qwirkle.AI/TileOnBoard.py
from Tile import Tile
class TileOnBoard(Tile):
    #def __init__(self, tile, coordinates):
    #    self.Shape = tile.Shape
    #    self.Color = tile.Color
    #    self.Coordinates = coordinates
    def __init__(self, tile):
        self.Shape = tile['shape']
        self.Color = tile['color']
        #self.Coordinates = coordinates

File: Qwirkle.Domain/ValueObjects/TileOnBoard.cs
namespace Qwirkle.Domain.ValueObjects;

public record TileOnBoard(Tile Tile, Coordinates Coordinates) : Tile(Tile)
{
    public TileOnBoard(TileColor color, TileShape shape, Coordinates coordinates) : this(new Tile(color, shape), coordinates) { }
    public static TileOnBoard From(Tile tile, Coordinates coordinates) => new(tile, coordinates);
    public static TileOnBoard From(TileOnBoard tile) => new(tile, tile.Coordinates);

    public Tile ToTile() => new(Color, Shape);
}